src/bin/pg_autoctl/ipaddr.c

```diff
@@ -79,31 +128,23 @@ fetchLocalIPAddress(char *localIpAddress, int size,
 		}
 		else
 		{
-			log_warn("Failed to connect: %m");
+			log_warn("Failed to connect to any of the IP addresses for "
+					 "monitor hostname \"%s\" and port %d",
```

Comment:
Maybe keep the `%m` at the end for debugging?

Reply:
It is now done at the connect call site. Also we do `getenv` in between so the errno has changed.